In [1]:
import pandas as pd
import numpy as np

In [2]:
df_1=pd.read_csv('Files used in current analysis/Linkedin_Job_Details_26-Sep-2023-to-26-Nov-2023_15875-jobs_685-pages.csv')
df_1

Job Title  \
0                              LinkedIn Data Extractor   
1                                 It Management Intern   
2                               SQL Database Developer   
3                      Business Intelligence Developer   
4                           Web Developer - Internship   
...                                                ...   
15870  Manager – Growth Chat & SMS (Digital Marketing)   
15871                         Release Support Engineer   
15872                  Information Security Specialist   
15873        FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
15874                              Community Associate   

                                                Job Info  \
0            EcomFleet · Pakistan (Remote)  \n1 week ago   
1      Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2      Murkez Technologies · Lahore, Punjab, Pakistan...   
3      DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4      ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                  ...   
15870  Bank Alfalah Limited · Karāchi, Sindh, Pakista...   
15871           Turing · Pakistan Reposted  \n1 week ago   
15872  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
15873  Level 3 BOS · Karachi Division, Sindh, Pakista...   
15874  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
0                                           Full-time   
1                                           Full-time   
2                                           Full-time   
3                                           Full-time   
4                                           Full-time   
...                                               ...   
15870  On-site\n\n \nFull-time\n\n \nMid-Senior level   
15871        Remote\n\n \nFull-time\n\n \nEntry level   
15872       On-site\n\n \nFull-time\n\n \nEntry level   
15873  On-site\n\n \nFull-time\n\n \nMid-Senior level   
15874                         On-site\n\n \nFull-time   

                                            Job Link  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
15870  https://www.linkedin.com/jobs/view/3767355025   
15871  https://www.linkedin.com/jobs/view/3766809535   
15872  https://www.linkedin.com/jobs/view/3772530977   
15873  https://www.linkedin.com/jobs/view/3771943212   
15874  https://www.linkedin.com/jobs/view/3767944844   

                                               employees  
0                                        11-50 employees  
1                                        11-50 employees  
2                                       51-200 employees  
3                                        11-50 employees  
4                    Mobeen Naqvi is hiring for this job  
...                                                  ...  
15870                   5,001-10,000 employees · Banking  
15871  1,001-5,000 employees · IT Services and IT Con...  
15872   51-200 employees · IT Services and IT Consulting  
15873  201-500 employees · Outsourcing and Offshoring...  
15874                                    11-50 employees  

[15875 rows x 5 columns]

In [3]:
df_2=pd.read_csv('Files used in current analysis/Linkedin_Job-Descriptions-company_details_26-Sep-2023-to-26-Nov-2023_15947-jobs_685-pages.csv')
df_2


Job Link  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
15942  https://www.linkedin.com/jobs/view/3767355025   
15943  https://www.linkedin.com/jobs/view/3766809535   
15944  https://www.linkedin.com/jobs/view/3772530977   
15945  https://www.linkedin.com/jobs/view/3771943212   
15946  https://www.linkedin.com/jobs/view/3767944844   

                                          Job Desception  \
0      About the job\n            \n\n \nAbout Us:We ...   
1      About the job\n            \n\n \nWe are looki...   
2      About the job\n            \n\n \nCompany Over...   
3      About the job\n            \n\n \nBI Developer...   
4      About the job\n            \n\n \nIt is a prog...   
...                                                  ...   
15942  About the job\n            \n \nThis role is p...   
15943  About the job\n            \n \nA U.S.-based c...   
15944                                      About the job   
15945                                      About the job   
15946                                      About the job   

                                            Company Info  
0      Advertising Services\n            \n          ...  
1      IT Services and IT Consulting\n            \n ...  
2      IT Services and IT Consulting\n            \n ...  
3      IT Services and IT Consulting\n            \n ...  
4                                                    NaN  
...                                                  ...  
15942                                                NaN  
15943  IT Services and IT Consulting\n            \n ...  
15944                                                NaN  
15945                                                NaN  
15946                                                NaN  

[15947 rows x 3 columns]

In [4]:
df_1['Job ID']=df_1['Job Link'].apply(lambda x:int(x.split('/')[5]))
df_2['Job ID']=df_2['Job Link'].apply(lambda x:int(x.split('/')[5]))

In [5]:
# df_1[df_1['Job ID'].isin(df_2['Job ID'])]
# df_2[df_2['Job ID'].isin(df_1['Job ID'])]
# pd.merge(df_1[df_1['Job ID'].isin(df_2['Job ID'])],df_2[df_2['Job ID'].isin(df_1['Job ID'])])
# pd.merge(df_1[df_1['Job ID'].isin(df_2['Job ID'])],df_2[df_2['Job ID'].isin(df_1['Job ID'])]).drop_duplicates(subset=['Job ID']).shape

In [5]:
df_3=pd.merge(df_1,df_2,on='Job ID',how='inner')
df_3

Job Title  \
0                              LinkedIn Data Extractor   
1                                 It Management Intern   
2                               SQL Database Developer   
3                      Business Intelligence Developer   
4                           Web Developer - Internship   
...                                                ...   
27639                                  Unity Developer   
27640  Manager – Growth Chat & SMS (Digital Marketing)   
27641                  Information Security Specialist   
27642        FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
27643                              Community Associate   

                                                Job Info  \
0            EcomFleet · Pakistan (Remote)  \n1 week ago   
1      Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2      Murkez Technologies · Lahore, Punjab, Pakistan...   
3      DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4      ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                  ...   
27639  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   
27640  Bank Alfalah Limited · Karāchi, Sindh, Pakista...   
27641  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
27642  Level 3 BOS · Karachi Division, Sindh, Pakista...   
27643  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
0                                           Full-time   
1                                           Full-time   
2                                           Full-time   
3                                           Full-time   
4                                           Full-time   
...                                               ...   
27639                         On-site\n\n \nFull-time   
27640  On-site\n\n \nFull-time\n\n \nMid-Senior level   
27641       On-site\n\n \nFull-time\n\n \nEntry level   
27642  On-site\n\n \nFull-time\n\n \nMid-Senior level   
27643                         On-site\n\n \nFull-time   

                                          Job Link_x  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
27639  https://www.linkedin.com/jobs/view/3767951886   
27640  https://www.linkedin.com/jobs/view/3767355025   
27641  https://www.linkedin.com/jobs/view/3772530977   
27642  https://www.linkedin.com/jobs/view/3771943212   
27643  https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
0                                        11-50 employees  3727054368   
1                                        11-50 employees  3723395858   
2                                       51-200 employees  3727034665   
3                                        11-50 employees  3727029273   
4                    Mobeen Naqvi is hiring for this job  3727034696   
...                                                  ...         ...   
27639                                     1-10 employees  3767951886   
27640                   5,001-10,000 employees · Banking  3767355025   
27641   51-200 employees · IT Services and IT Consulting  3772530977   
27642  201-500 employees · Outsourcing and Offshoring...  3771943212   
27643                                    11-50 employees  3767944844   

                                          Job Link_y  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
27639  https

In [7]:
# why df_3 have 27644 rows

In [8]:
df_1.drop_duplicates(subset=['Job ID']).shape

(12010, 6)

In [9]:
df_2.drop_duplicates(subset=['Job ID']).shape

(12321, 4)

In [10]:
df_3.drop_duplicates(subset=['Job ID']).shape

(12010, 9)

In [6]:
df_1.isnull().sum()

Job Title    288
Job Info     288
Job Type     302
Job Link       0
employees    303
Job ID         0
dtype: int64

In [12]:
df_2.isnull().sum()

Job Link             0
Job Desception     315
Company Info      4084
Job ID               0
dtype: int64

In [7]:
df_3.isnull().sum()

Job Title          348
Job Info           348
Job Type           362
Job Link_x           0
employees          363
Job ID               0
Job Link_y           0
Job Desception     385
Company Info      6478
dtype: int64

***why there are 348 Null values in Job Title instead of 288***

In [14]:
df_3.columns

Index(['Job Title', 'Job Info', 'Job Type', 'Job Link_x', 'employees',
       'Job ID', 'Job Link_y', 'Job Desception', 'Company Info'],
      dtype='object')

In [15]:
df_1.dropna(inplace=True)
df_2.dropna(subset=['Job Desception'],inplace=True)
df_3.dropna(subset=['Job Title','Job Info','Job Type','employees'],inplace=True) # i want to delete specific columns

In [16]:
df_1.shape,df_2.shape,df_3.shape

((15572, 6), (15632, 4), (27281, 9))

In [17]:
df_1=df_1.drop_duplicates(subset=['Job ID'])
df_2=df_2.drop_duplicates(subset=['Job ID'])
df_3=df_3.drop_duplicates(subset=['Job ID'])

In [18]:
df_1.shape,df_2.shape,df_3.shape

((11744, 6), (12037, 4), (11744, 9))

In [19]:
12010-363

11647

In [20]:
df_3

Job Title  \
0                              LinkedIn Data Extractor   
1                                 It Management Intern   
2                               SQL Database Developer   
3                      Business Intelligence Developer   
4                           Web Developer - Internship   
...                                                ...   
27639                                  Unity Developer   
27640  Manager – Growth Chat & SMS (Digital Marketing)   
27641                  Information Security Specialist   
27642        FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
27643                              Community Associate   

                                                Job Info  \
0            EcomFleet · Pakistan (Remote)  \n1 week ago   
1      Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2      Murkez Technologies · Lahore, Punjab, Pakistan...   
3      DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4      ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                  ...   
27639  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   
27640  Bank Alfalah Limited · Karāchi, Sindh, Pakista...   
27641  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
27642  Level 3 BOS · Karachi Division, Sindh, Pakista...   
27643  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
0                                           Full-time   
1                                           Full-time   
2                                           Full-time   
3                                           Full-time   
4                                           Full-time   
...                                               ...   
27639                         On-site\n\n \nFull-time   
27640  On-site\n\n \nFull-time\n\n \nMid-Senior level   
27641       On-site\n\n \nFull-time\n\n \nEntry level   
27642  On-site\n\n \nFull-time\n\n \nMid-Senior level   
27643                         On-site\n\n \nFull-time   

                                          Job Link_x  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
27639  https://www.linkedin.com/jobs/view/3767951886   
27640  https://www.linkedin.com/jobs/view/3767355025   
27641  https://www.linkedin.com/jobs/view/3772530977   
27642  https://www.linkedin.com/jobs/view/3771943212   
27643  https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
0                                        11-50 employees  3727054368   
1                                        11-50 employees  3723395858   
2                                       51-200 employees  3727034665   
3                                        11-50 employees  3727029273   
4                    Mobeen Naqvi is hiring for this job  3727034696   
...                                                  ...         ...   
27639                                     1-10 employees  3767951886   
27640                   5,001-10,000 employees · Banking  3767355025   
27641   51-200 employees · IT Services and IT Consulting  3772530977   
27642  201-500 employees · Outsourcing and Offshoring...  3771943212   
27643                                    11-50 employees  3767944844   

                                          Job Link_y  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
27639  https

In [21]:
df_4=df_3.dropna(subset=['Job Desception'])
df_4

Job Title  \
0                              LinkedIn Data Extractor   
1                                 It Management Intern   
2                               SQL Database Developer   
3                      Business Intelligence Developer   
4                           Web Developer - Internship   
...                                                ...   
27639                                  Unity Developer   
27640  Manager – Growth Chat & SMS (Digital Marketing)   
27641                  Information Security Specialist   
27642        FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
27643                              Community Associate   

                                                Job Info  \
0            EcomFleet · Pakistan (Remote)  \n1 week ago   
1      Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2      Murkez Technologies · Lahore, Punjab, Pakistan...   
3      DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4      ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                  ...   
27639  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   
27640  Bank Alfalah Limited · Karāchi, Sindh, Pakista...   
27641  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
27642  Level 3 BOS · Karachi Division, Sindh, Pakista...   
27643  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
0                                           Full-time   
1                                           Full-time   
2                                           Full-time   
3                                           Full-time   
4                                           Full-time   
...                                               ...   
27639                         On-site\n\n \nFull-time   
27640  On-site\n\n \nFull-time\n\n \nMid-Senior level   
27641       On-site\n\n \nFull-time\n\n \nEntry level   
27642  On-site\n\n \nFull-time\n\n \nMid-Senior level   
27643                         On-site\n\n \nFull-time   

                                          Job Link_x  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
27639  https://www.linkedin.com/jobs/view/3767951886   
27640  https://www.linkedin.com/jobs/view/3767355025   
27641  https://www.linkedin.com/jobs/view/3772530977   
27642  https://www.linkedin.com/jobs/view/3771943212   
27643  https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
0                                        11-50 employees  3727054368   
1                                        11-50 employees  3723395858   
2                                       51-200 employees  3727034665   
3                                        11-50 employees  3727029273   
4                    Mobeen Naqvi is hiring for this job  3727034696   
...                                                  ...         ...   
27639                                     1-10 employees  3767951886   
27640                   5,001-10,000 employees · Banking  3767355025   
27641   51-200 employees · IT Services and IT Consulting  3772530977   
27642  201-500 employees · Outsourcing and Offshoring...  3771943212   
27643                                    11-50 employees  3767944844   

                                          Job Link_y  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
27639  https

In [22]:
df_4.shape

(11708, 9)

In [23]:
df_4_uncomplete_desc_IDs=[]
for i in range(0,len(df_4.iloc[:])):
    if len(df_4.iloc[i]['Job Desception'])<20:
        df_4_uncomplete_desc_IDs.append(df_4.iloc[i]['Job ID'])    

In [24]:
df_4_uncomplete_desc_IDs
len(df_4_uncomplete_desc_IDs)

1125

In [25]:
df_4_uncomp_desc=df_4[df_4['Job ID'].isin(df_4_uncomplete_desc_IDs)]
df_4_uncomp_desc

Job Title  \
120             Digital Marketing Strategist and Planner   
328                        Software Engineer-Integration   
801                                      DevOps Engineer   
842                              Real Estate Sales Agent   
849                                Google Ads Specialist   
...                                                  ...   
27638  Senior Consultant- Business Consulting- Intern...   
27639                                    Unity Developer   
27641                    Information Security Specialist   
27642          FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
27643                                Community Associate   

                                                Job Info  \
120    Prime Human Resource Services (Pvt) Ltd · Karā...   
328    People (Professional Employers Pvt Ltd) · Isla...   
801    TecBrix · Lahore, Punjab, Pakistan (Remote)  \...   
842    Fajar Real Estate · Lahore, Punjab, Pakistan (...   
849    RT WEB Solutions · Lahore District, Punjab, Pa...   
...                                                  ...   
27638  EY in Pakistan  · Karachi Division, Sindh, Pak...   
27639  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   
27641  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
27642  Level 3 BOS · Karachi Division, Sindh, Pakista...   
27643  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
120                                         Full-time   
328                                         Full-time   
801                                         Full-time   
842                                         Full-time   
849                                         Full-time   
...                                               ...   
27638  Hybrid\n\n \nInternship\n\n \nMid-Senior level   
27639                         On-site\n\n \nFull-time   
27641       On-site\n\n \nFull-time\n\n \nEntry level   
27642  On-site\n\n \nFull-time\n\n \nMid-Senior level   
27643                         On-site\n\n \nFull-time   

                                              Job Link_x  \
120        https://www.linkedin.com/jobs/view/3727011545   
328        https://www.linkedin.com/jobs/view/3727002795   
801    https://www.linkedin.com/jobs/view/3727012776/...   
842    https://www.linkedin.com/jobs/view/3727005694/...   
849    https://www.linkedin.com/jobs/view/3727058923/...   
...                                                  ...   
27638      https://www.linkedin.com/jobs/view/3767944882   
27639      https://www.linkedin.com/jobs/view/3767951886   
27641      https://www.linkedin.com/jobs/view/3772530977   
27642      https://www.linkedin.com/jobs/view/3771943212   
27643      https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
120                                    10,001+ employees  3727011545   
328                               5,001-10,000 employees  3727002795   
801                                      11-50 employees  3727012776   
842                                      11-50 employees  3727005694   
849                                      11-50 employees  3727058923   
...                                                  ...         ...   
27638                 1,001-5,000 employees · Accounting  3767944882   
27639                                     1-10 employees  3767951886   
27641   51-200 employees · IT Services and IT Consulting  3772530977   
27642  201-500 employees · Outsourcing and Offshoring...  3771943212   
27643                                    11-50 employees  3767944844   

                                          Job Link_y Job Desception  \
120    https://www.linkedin.com/jobs/view/3727011545  About the job   
328    https://www.linkedin.com/jobs/view/3727002795  About the job   
801    https://www.linkedin.com/jobs/view/3727012776  About the job   
842    https://www.linkedin.com/jobs/view/37270056

In [26]:
df_5=df_4.drop(df_4_uncomp_desc.index)
df_5

Job Title  \
0                                LinkedIn Data Extractor   
1                                   It Management Intern   
2                                 SQL Database Developer   
3                        Business Intelligence Developer   
4                             Web Developer - Internship   
...                                                  ...   
27634                 Technical Sales Support Specialist   
27635                           YouTube Channel Producer   
27636  Social Media Content Creator - Night Shift wit...   
27637                                Design Sales agents   
27640    Manager – Growth Chat & SMS (Digital Marketing)   

                                                Job Info  \
0            EcomFleet · Pakistan (Remote)  \n1 week ago   
1      Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2      Murkez Technologies · Lahore, Punjab, Pakistan...   
3      DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4      ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                  ...   
27634  BUZZMAX.CA · Karāchi, Sindh, Pakistan  \n1 wee...   
27635  Senew Tech · Lahore District, Punjab, Pakistan...   
27636  Dymentions Pakistan · Karāchi, Sindh, Pakistan...   
27637  Self-employed · Pakistan  \n1 week ago\n\n · 1...   
27640  Bank Alfalah Limited · Karāchi, Sindh, Pakista...   

                                             Job Type  \
0                                           Full-time   
1                                           Full-time   
2                                           Full-time   
3                                           Full-time   
4                                           Full-time   
...                                               ...   
27634                         On-site\n\n \nFull-time   
27635                         On-site\n\n \nFull-time   
27636       On-site\n\n \nFull-time\n\n \nEntry level   
27637                          Remote\n\n \nPart-time   
27640  On-site\n\n \nFull-time\n\n \nMid-Senior level   

                                          Job Link_x  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
27634  https://www.linkedin.com/jobs/view/3765162094   
27635  https://www.linkedin.com/jobs/view/3767643794   
27636  https://www.linkedin.com/jobs/view/3769970428   
27637  https://www.linkedin.com/jobs/view/3766852400   
27640  https://www.linkedin.com/jobs/view/3767355025   

                                   employees      Job ID  \
0                            11-50 employees  3727054368   
1                            11-50 employees  3723395858   
2                           51-200 employees  3727034665   
3                            11-50 employees  3727029273   
4        Mobeen Naqvi is hiring for this job  3727034696   
...                                      ...         ...   
27634                        11-50 employees  3765162094   
27635                    501-1,000 employees  3767643794   
27636   11-50 employees · Marketing Services  3769970428   
27637  Shayan Tauqeer is hiring for this job  3766852400   
27640       5,001-10,000 employees · Banking  3767355025   

                                          Job Link_y  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
27634  https://www.linkedin.com/jobs/view/3765162094   
27635  https://www.linkedin.com/jobs/view/3767643794   
27636  https://www.lin

In [27]:
# df_2[df_2['Job ID']==3769970428] for verification

#### using commen keywords used in IT and Tech

In [28]:
Software_Eng_keywords_all=['Software Engineer','Software Developer','Software/Application Developer','Software Development',
                           'Software Quality Assurance','Software Architect','Software Project Manager','QA',
                           'Software Test Engineer','Software Engineering','Software Trainee Engineer',
                           'Software Document Writer','Software Delivery Engineer','Software Support Manager',
                           'Product Specialist - Software Solutions',
                           'Software Services Lead','Software Project Coordinator',
                           'SDDC Consultant (Software-Defined Data Center)','Software Solutions Architect',
                           'Hardware/Software Tester','Software Support Engineer', 'Engineering Manager, GFI Software',
                           'Fellow of GenAI Software Excellence','Pre-Sales Software']
Software_Eng_keywords_all
len(Software_Eng_keywords_all)

filtered_df_3_title = df_3[df_3['Job Title'].apply(lambda title: any(keyword.lower() in title.lower() for keyword in Software_Eng_keywords_all))]
filtered_df_3_title

Job Title  \
122                                    Software Engineer   
130                Software Engineer - Product Operation   
168                                   SOFTWARE DEVELOPER   
185                                    Software Engineer   
198                  Software Quality Assurance Engineer   
...                                                  ...   
27588  Lead Software Engineer, Trilogy (Remote) - $10...   
27598  Senior C++ Software Engineer, GFI Software (Re...   
27615                                        QA Engineer   
27627  Principal Software Engineer, IgniteTech (Remot...   
27632    Software Development Support Engineer (L2 & L3)   

                                                Job Info  \
122    Jin Technologies Private Limited · Islamabad, ...   
130    i2c Inc. · Lahore, Punjab, Pakistan (On-site) ...   
168    360 TECHNOLOGIES · Islamabad, Islāmābād, Pakis...   
185    Telenor · Islamabad, Islāmābād, Pakistan (On-s...   
198    Brickclay · Faisalabad, Punjab, Pakistan (On-s...   
...                                                  ...   
27588  Crossover  · Lahore, Punjab, Pakistan Reposted...   
27598  Crossover  · Lahore, Punjab, Pakistan Reposted...   
27615  NewRich Network · Lahore, Punjab, Pakistan  \n...   
27627  Crossover  · Karāchi, Sindh, Pakistan Reposted...   
27632  the ENTERTAINER · Lahore, Punjab, Pakistan  \n...   

                                             Job Type  \
122                                         Full-time   
130                                         Full-time   
168                                         Full-time   
185                           Full-time · Entry level   
198                                         Full-time   
...                                               ...   
27588   Remote\n\n \nFull-time\n\n \nMid-Senior level   
27598   Remote\n\n \nFull-time\n\n \nMid-Senior level   
27615        Remote\n\n \nFull-time\n\n \nEntry level   
27627   Remote\n\n \nFull-time\n\n \nMid-Senior level   
27632  On-site\n\n \nFull-time\n\n \nMid-Senior level   

                                          Job Link_x  \
122    https://www.linkedin.com/jobs/view/3727035963   
130    https://www.linkedin.com/jobs/view/3727020559   
168    https://www.linkedin.com/jobs/view/3727057491   
185    https://www.linkedin.com/jobs/view/3723693959   
198    https://www.linkedin.com/jobs/view/3727007394   
...                                              ...   
27588  https://www.linkedin.com/jobs/view/3766612589   
27598  https://www.linkedin.com/jobs/view/3766356253   
27615  https://www.linkedin.com/jobs/view/3772083919   
27627  https://www.linkedin.com/jobs/view/3766351795   
27632  https://www.linkedin.com/jobs/view/3763867444   

                                               employees      Job ID  \
122                                     51-200 employees  3727035963   
130                                1,001-5,000 employees  3727020559   
168                                     51-200 employees  3727057491   
185               10,001+ employees · Telecommunications  3723693959   
198                                     51-200 employees  3727007394   
...                                                  ...         ...   
27588  5,001-10,000 employees · IT Services and IT Co...  3766612589   
27598  5,001-10,000 employees · IT Services and IT Co...  3766356253   
27615     11-50 employees · Online Audio and Video Media  3772083919   
27627  5,001-10,000 employees · IT Services and IT Co...  3766351795   
27632           201-500 employees · Software Development  3763867444   

                                          Job Link_y  \
122    https://www.linkedin.com/jobs/view/3727035963   
130    https://www.linkedin.com/jobs/view/3727020559   
168    https://www.linkedin.com/jobs/view/3727057491   
185    https://www.linkedin.com/jobs/view/3723693959   
198    https://www.linkedin.com/jobs/view/3727007394   
...                                           

In [29]:
filtered_df_3_title['Job Title'].unique()
# len(filtered_df_3_title['Job Title'].unique())

array(['Software Engineer', 'Software Engineer - Product Operation',
       'SOFTWARE DEVELOPER', 'Software Quality Assurance Engineer',
       'Senior Software Engineer', 'Software Engineer-Integration',
       'Professional, QA',
       'AI-Augmented Software Developer, Trilogy (Remote) - $200,000/year USD',
       'Senior Software Quality Assurance Engineer',
       'Software Development Engineer in Test (SDET)',
       'Software Developer', 'Software Engineer - Marketing Platform',
       'Senior QA Engineer',
       'Senior Software Development Project Manager',
       'Software Quality Assurance Analyst',
       'Software Architect Lead Developer',
       'Software Engineer (.Net/MVC)', 'Principal Software Engineer',
       'Senior Software Engineer (.NET/ React Js)',
       'Principal Software Engineer Full Stack PHP',
       'Senior Java Software Engineer',
       'Team Lead IT Software Development (m/f/d)',
       'AI-Augmented Software Developer, Skyvera (Remote) - $200,000/y

In [30]:
titles_to_drop_old=['Civil Engineers (Roads, QA/QC) QS','Business Development Manager (Software Development Services)']

In [31]:
Un_QA_titles=['Technical Advisor - Family Planning - Qadam Ba Qadam Project','Program & Administrative Assistant - Qadam Ba Qadam Project',
              'Administrative Assistant - Qadam Ba Qadam Project',
              'Social Behavior Change Communication Specialist - Qadam Ba Qadam Project',
              'Employee Service Executive - Qasba Gujrat (H-164)','Professional, QA']
Un_QA_titles

['Technical Advisor - Family Planning - Qadam Ba Qadam Project',
 'Program & Administrative Assistant - Qadam Ba Qadam Project',
 'Administrative Assistant - Qadam Ba Qadam Project',
 'Social Behavior Change Communication Specialist - Qadam Ba Qadam Project',
 'Employee Service Executive - Qasba Gujrat (H-164)',
 'Professional, QA']

In [32]:
titles_to_drop=titles_to_drop_old+Un_QA_titles
titles_to_drop
len(titles_to_drop)

8

In [34]:
df_3_Software_Eng=filtered_df_3_title[~filtered_df_3_title['Job Title'].isin(titles_to_drop)]
df_3_Software_Eng
len(df_3_Software_Eng)
len(df_3_Software_Eng['Job Title'].unique())

304

In [36]:
df_3_Software_Eng.to_csv('Filltered_Files/software_engineer_26_sep_to_26_nov_735_jobs_304_titles_v5.csv',index=False)